app/app.js

```diff
@@ -195,13 +215,22 @@ function initializeConnections() {
       }
     })
     .finally(() => {
-      state.ready = Object.values(state.connections).every((x) => x);
+      state.ready = state.connections.data; // only need db running
       if (state.ready) {
         log.info('Service ready to accept traffic', {
           function: 'initializeConnections',
         });
         // Start periodic 10 second connection probe check
         probeId = setInterval(checkConnections, 10000);
+      } else {
+        log.error(`Service not ready to accept traffic`, {
+          function: 'initializeConnections',
+        });
+        log.error(`Database connected = ${state.connections.data}`, { function: 'initializeConnections' });
+        log.error(`EventStreamService connected = ${state.connections.eventStreamService}`, { function: 'initializeConnections' });
+
+        process.exitCode = 1;
+        shutdown();
```

Comment:
What is the effect of this shutdown? Instead of the pod waiting for the database to be reachable, the pod now goes into a restart loop? Or were the pods not connecting to the database once it was available?

Reply:
it's basically copied from the same situation as above it - that the db connection isn't active. we didn't have any logic if the state wasn't ready. I'll remove it (so basically like it was before) and we can address it if needed.